providers/azure.go

```diff
@@ -220,6 +223,35 @@ func (p *AzureProvider) EnrichSession(ctx context.Context, session *sessions.Ses
 	return nil
 }
 
+func (p *AzureProvider) setUser(ctx context.Context, s *sessions.SessionState) {
+	header := makeAzureHeader(s.AccessToken)
+	endpoint := p.Data().ValidateURL.String()
+	if len(header) == 0 {
+		params := url.Values{"access_token": {s.AccessToken}}
+		if hasQueryParams(endpoint) {
+			endpoint = endpoint + "&" + params.Encode()
+		} else {
+			endpoint = endpoint + "?" + params.Encode()
+		}
+	}
+
+	result := requests.New(endpoint).
+		WithContext(ctx).
+		WithHeaders(header).
+		Do()
+	if result.Error() != nil {
+		logger.Errorf("GET %s", stripToken(endpoint))
+	}
+
+	logger.Printf("%d GET %s %s", result.StatusCode(), stripToken(endpoint), result.Body())
+	var jsonMap map[string]interface{}
+	err := json.Unmarshal(result.Body(), &jsonMap)
+	if err != nil {
+		return
+	}
```

Comment:
I would suggest a trim down of the code. Furthermore I would prefer if you use a fixed struct instead of `map[string]interface{}` because the current implementation will cause oauth2-proxy to panic and crash if for spme reason the `name` key does not exist and you will try to cast `nil` to `string`.

> panic: interface conversion: interface {} is nil, not string

Have a look at how it has been done in `redeemRefreshToken`.


```suggestion
  
	var jsonResponse map[string]interface{}

	err := requests.New(endpoint).
		WithContext(ctx).
		WithHeaders(header).
		Do().
		UnmarshalInto(&jsonResponse)
	if err != nil {
		logger.Printf("GET %s %w", stripToken(endpoint), err)
		return err
	}
	
	s.User = jsonResponse["name"].(string)
```